### ISSUE TITLE
Background Usage

### ISSUE THREAD

**Issue by degnarraer** [andresbrun/ABMSoundCloudAPI#9]
I would like to use this API while my app is in the background.  However, I get this error whenever my app try to download the next song and it is in the background:

Unable to fetch credential with identifier "SoundClount_Crendentials" (Error -25308)

**Comment by andresbrun** [comment]
Hi degnarraer, sorry for the delay. This error is directly related with AFOAuthCredential, the networking library I am using. I have to investigate a little bit more why it is not working on background.

**Comment by andresbrun** [comment]
degnarraer, I updated AFOAuthCredential to last version (3.0.0+). Can you try with this change and let me know if the error persists?

**Comment by degnarraer** [comment]
I will let you know

**Comment by degnarraer** [comment]
Did not appear to resolve the issue

**Comment by degnarraer** [comment]
download are allowed for ~30 seconds once the app is put into the background, after that, they no longer are allowed

**Comment by degnarraer** [comment]
I made some changes that seem to work.  I stored off the credentials local to your api to avoid calling into AFOuth....However, once they expire... they will likely not be renewed on the background.  I am not that knowledgeable about how all this works, so I am excited to know what you find out.  While in background mode, [AFOAuthCredential retrieveCredentialWithIdentifier:PROVIDER_IDENTIFIER]; is the function that fails.

here is a summary of the changes.

//
//  SoundCloudPort.h
//  ABMSoundCloudAPI
//
//  Created by AndresBrun on 01/10/14.
//  Copyright (c) 2014 Brun's Software. All rights reserved.
//

@class AFOAuthCredential;

@interface SoundCloudPort : NSObject{
    AFOAuthCredential *storedCredentials;
}
...
...
...

'- (AFOAuthCredential *)credentials {
    return [self getStoredCredentials];
}
'- (AFOAuthCredential *)getStoredCredentials {
    if (storedCredentials && ![storedCredentials isExpired]) {
        return storedCredentials;
    }
    else{
        storedCredentials = [AFOAuthCredential retrieveCredentialWithIdentifier:PROVIDER_IDENTIFIER];
    }
    return storedCredentials;
}
